test/groovy/org/ods/services/GitServiceSpec.groovy

```diff
@@ -18,5 +18,73 @@ class GitServiceSpec extends SpecHelper {
         then:
         releaseBranch == "release/0.0.1"
     }
-}
 
+    def "git commit with skip pattern: [ci skip]"() {
+        given:
+        def gitCommitMessage = "docs: update README [ci skip]"
+        def script = new PipelineScript()
+        def service = new GitService(script, new Logger(script, false))
+
+        when:
+        def isSkippingCommitMessage = service.isCiSkipInCommitMessage(gitCommitMessage)
+
+        then:
+        isSkippingCommitMessage == true
+    }
+
+    def "git commit with skip pattern: [skip ci]"() {
+        given:
+        def gitCommitMessage = "docs: update README [skip ci]"
+        def script = new PipelineScript()
+        def service = new GitService(script, new Logger(script, false))
+
+        when:
+        def isSkippingCommitMessage = service.isCiSkipInCommitMessage(gitCommitMessage)
+
+        then:
+        isSkippingCommitMessage == true
+    }
+
+    def "git commit with skip pattern: ***NO_CI***"() {
+        given:
+        def gitCommitMessage = "docs: update README ***NO_CI***"
+        def script = new PipelineScript()
+        def service = new GitService(script, new Logger(script, false))
+
+        when:
+        def isSkippingCommitMessage = service.isCiSkipInCommitMessage(gitCommitMessage)
+
+        then:
+        isSkippingCommitMessage == true
+    }
+
+    def "git commit without skip pattern"() {
+        given:
+        def gitCommitMessage = "docs: update README"
+        def script = new PipelineScript()
+        def service = new GitService(script, new Logger(script, false))
+
+        when:
+        def isSkippingCommitMessage = service.isCiSkipInCommitMessage(gitCommitMessage)
+
+        then:
+        isSkippingCommitMessage == false
+    }
+
+    def "git commit multiline including skip pattern"() {
+        given:
+        def gitCommitMessage = """docs: update README
+
+        - typo
+        - [ci skip]
+        """
+        def script = new PipelineScript()
+        def service = new GitService(script, new Logger(script, false))
+
+        when:
+        def isSkippingCommitMessage = service.isCiSkipInCommitMessage(gitCommitMessage)
+
+        then:
+        isSkippingCommitMessage == false
+    }
```

Comment:
I'm fine with this, but note that there is a way to condense this drastically, and make it easier to understand what you are actually testing ... using `where` "tables" ... see e.g. https://github.com/opendevstack/ods-jenkins-shared-library/blob/master/test/groovy/org/ods/services/OpenShiftServiceSpec.groovy#L12-L31

Reply:
Nice! I was not aware of this before and adjusted the test cases respectively.